statsd/options.go

```diff
@@ -297,3 +300,15 @@ func WithTelemetryAddr(addr string) Option {
 		return nil
 	}
 }
+
+// WithoutOriginDetection disables the client origin detection.
+// When enabled, the client tries to discover its container ID and sends it to the Agent
+// to enrich the metrics with container tags.
```

Comment:
We also need a `WithOriginDetection`, even if it's enabled by default. Some users have helpers that set some default settings and then each team/project can overwrite them. So this could be disabled by default and then enabled manually for a specific project. We would need tests for both option too.

Also, do we have a documentation we could link on how we discover the container ID ? If so, it would be nice to link it here. If not we should explain a bit more how we discover it.

Lastly, let's explain that it does not work on Windows and how `DD_ENTITY_ID` and `DD_ORIGIN_DETECTION_ENABLED` impact it.